scripts/restore.sh

```diff
@@ -292,10 +292,15 @@ handle_session_0() {
 	fi
 }
 
-restore_pane_layout_for_each_window() {
+restore_pane_layout_and_autonaming_for_each_window() {
 	\grep '^window' $(last_resurrect_file) |
-		while IFS=$d read line_type session_name window_number window_active window_flags window_layout; do
+		while IFS=$d read line_type session_name window_number window_active window_flags window_layout automatic_rename; do
 			tmux select-layout -t "${session_name}:${window_number}" "$window_layout"
+			if [ "${automatic_rename}" = ":" ]; then
+				tmux set-option -u -t "${session_name}:${window_number}" automatic-rename
```

Comment:
I guess the only problem here is:

- We're first setting window name for each window in `restore_pane`

https://github.com/tmux-plugins/tmux-resurrect/blob/716b958145e2709e9fd3639ff24e1d1e4f26ff77/scripts/restore.sh#L180-L211

- Then `automatic-rename` is unset here (if needed).

That seems like duplicate work. As mentioned [here](https://github.com/tmux-plugins/tmux-resurrect/pull/222#issuecomment-892447307) I think we'll want to change where `window_name` is saved and restored.